sdk/src/main/java/io/dapr/client/DaprClientHttp.java

```diff
@@ -657,6 +660,110 @@ public Mono<Map<String, Map<String, String>>> getBulkSecret(GetBulkSecretRequest
         .map(m -> (Map<String, Map<String, String>>) m);
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Mono<Boolean> tryLock(LockRequest request) {
+    try {
+      final String stateStoreName = request.getStoreName();
+      final String resourceId = request.getResourceId();
+      final String lockOwner = request.getLockOwner();
+      final Integer expiryInSeconds = request.getExpiryInSeconds();
+
+      if ((stateStoreName == null) || (stateStoreName.trim().isEmpty())) {
+        throw new IllegalArgumentException("State store name cannot be null or empty.");
+      }
+      if (resourceId == null || resourceId.isEmpty()) {
+        throw new IllegalArgumentException("ResourceId cannot be null or empty.");
+      }
+      if (lockOwner == null || lockOwner.isEmpty()) {
+        throw new IllegalArgumentException("LockOwner cannot be null or empty.");
+      }
+      if (expiryInSeconds < 0) {
+        throw new IllegalArgumentException("ExpiryInSeconds cannot be negative.");
+      }
+
+
+      Map<String, Object> jsonMap = new HashMap<>();
+      jsonMap.put("resourceId", resourceId);
+      jsonMap.put("lockOwner", lockOwner);
+      jsonMap.put("expiryInSeconds", expiryInSeconds);
+
+      byte[] requestBody = INTERNAL_SERIALIZER.serialize(jsonMap);
+
+      String[] pathSegments = new String[]{DaprHttp.API_VERSION, "lock", stateStoreName};
+
+      return Mono.subscriberContext().flatMap(
+              context -> this.client
+                      .invokeApi(DaprHttp.HttpMethods.POST.name(), pathSegments, null, requestBody, null, context)
+      ).flatMap(response -> {
+        try {
+          Map m = INTERNAL_SERIALIZER.deserialize(response.getBody(), Map.class);
+          if (m == null) {
+            return Mono.just(Boolean.FALSE);
+          }
+          return Mono.just(Boolean.valueOf((String) m.get("success")));
+        } catch (Exception ex) {
+          return DaprException.wrapMono(ex);
+        }
+      });
+    } catch (Exception ex) {
+      return DaprException.wrapMono(ex);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Mono<UnlockResponseStatus> unLock(UnLockRequest request) {
+    try {
+      final String stateStoreName = request.getStoreName();
+      final String resourceId = request.getResourceId();
+      final String lockOwner = request.getLockOwner();
+
+      if ((stateStoreName == null) || (stateStoreName.trim().isEmpty())) {
+        throw new IllegalArgumentException("State store name cannot be null or empty.");
+      }
+      if (resourceId == null || resourceId.isEmpty()) {
+        throw new IllegalArgumentException("ResourceId cannot be null or empty.");
+      }
+      if (lockOwner == null || lockOwner.isEmpty()) {
+        throw new IllegalArgumentException("LockOwner cannot be null or empty.");
+      }
+
+
+      Map<String, Object> jsonMap = new HashMap<>();
```

Comment:
Make the UnlockRequest itself a json serializable object and have only required fields in json form using Jackson annotations?

Reply:
Also magic strings (see above request)